Sources/PromotedCore/AnomalyHandling/AnomalyModalViewController.swift

```diff
@@ -0,0 +1,200 @@
+  func anomalyModalVCDidDismiss(
+    _ vc: AnomalyModalViewController,
+    shouldShowAgain: Bool
+  )
+}
+
+/**
+ Presents a modal with detailed error message. Clients should not use this
+ class directly.
+
+ Although this class is intended only for internal use in the Promoted metrics
+ library, it is declared public so that ReactNativeMetrics can show the modal
+ for module initialization issues.
+ */
+public class AnomalyModalViewController: UIViewController {
+
+  weak var delegate: AnomalyModalViewControllerDelegate?
+
+  private let partner: String
+  private let contactInfo: [String]
+  private let anomalyType: AnomalyType
+
+  private var shouldShowAgain: Bool
+
+  init(
+    partner: String,
+    contactInfo: [String],
+    anomalyType: AnomalyType,
+    delegate: AnomalyModalViewControllerDelegate?
+  ) {
+    self.partner = partner
+    self.contactInfo = contactInfo
+    self.anomalyType = anomalyType
+    self.delegate = delegate
+    self.shouldShowAgain = true
+    super.init(nibName: nil, bundle: nil)
+    if #available(iOS 13.0, *) {
+      self.isModalInPresentation = true  // Prevent dragging down to dismiss
+    }
+  }
+
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  public override func viewWillAppear(_ animated: Bool) {
+    super.viewWillAppear(animated)
+
+    guard let view = self.view else { return }
+    let backdropWidth = view.bounds.width - 40
+    let textWidth = backdropWidth - 40
+    let textLayoutFrame = CGRect(x: 0, y: 0, width: textWidth, height: 0)
+
+    let blurEffect = UIBlurEffect(style: .dark)
+    let backdrop = UIVisualEffectView(effect: blurEffect)
+    backdrop.clipsToBounds = true
+    backdrop.layer.cornerRadius = 20
+    backdrop.translatesAutoresizingMaskIntoConstraints = false
+    view.addSubview(backdrop)
+
+    let titleLabel = UILabel(frame: textLayoutFrame)
+    titleLabel.font = .boldSystemFont(ofSize: 16)
+    titleLabel.text = "Promoted.ai Logging Issue"
+    titleLabel.textColor = .white
+    titleLabel.translatesAutoresizingMaskIntoConstraints = false
+    backdrop.contentView.addSubview(titleLabel)
+
+    let explanationLabel = UILabel(frame: textLayoutFrame)
+    explanationLabel.numberOfLines = 0  // Use as many lines as needed.
+    explanationLabel.text = """
+    \(anomalyType.debugDescription)
+
+    Error code: \(anomalyType.rawValue)
+    """
+    explanationLabel.textColor = .white
+    explanationLabel.translatesAutoresizingMaskIntoConstraints = false
+    backdrop.contentView.addSubview(explanationLabel)
+
+    let helpLabel = UILabel(frame: textLayoutFrame)
+    helpLabel.font = .systemFont(ofSize: 14)
+    helpLabel.numberOfLines = 0  // Use as many lines as needed.
+    helpLabel.text = """
+    If this issue is released to production, it WILL impair Promoted Delivery and possibly affect revenue at \(partner). Please verify any local changes carefully before merging.
```

Comment:
Side - how efficiently do dev vs prod builds work on mobile?  E.g. will these strings be included in their prod builds?

Reply:
For now, they will. But I'm going to strip them out in a subsequent PR.